src/queue.rs

```diff
@@ -142,7 +174,8 @@ impl<M: GuestAddressSpace> DescriptorChain<M> {
         }
     }
 
-    fn checked_new(
+    /// Create a new DescriptorChain instance.
+    pub fn checked_new(
```

Comment:
Why does this need to be `pub`? Can it be `pub(crate)` instead?

Reply:
fixed